ignite/cmd/plugin_test.go

```diff
@@ -242,6 +262,114 @@ ignite
 	}
 }
 
+func TestLinkPluginHooks(t *testing.T) {
+	buildRootCmd := func() *cobra.Command {
+		var (
+			rootCmd = &cobra.Command{
+				Use: "ignite",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+			scaffoldCmd = &cobra.Command{
+				Use: "scaffold",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+			scaffoldChainCmd = &cobra.Command{
+				Use: "chain",
+				RunE: func(*cobra.Command, []string) error {
+					return nil
+				},
+			}
+		)
+		scaffoldCmd.AddCommand(scaffoldChainCmd)
+		rootCmd.AddCommand(scaffoldCmd)
+		return rootCmd
+	}
+	tests := []struct {
+		name            string
+		pluginInterface pluginInterface
+		shouldError     bool
+	}{
+		{
+			name: "error: invalid command path",
+			pluginInterface: pluginInterface{
+				commands: []plugin.Command{
+					{
+						Use: "foo",
+					},
+				},
+				hooks: []plugin.Hook{
+					{
+						Name:        "test-hook-1",
+						PlaceHookOn: "ignite scaffold",
+					},
+				},
+			},
+			shouldError: false,
+		},
+		{
+			name: "ok: link foo at subcommand",
+			pluginInterface: pluginInterface{
+				hooks: []plugin.Hook{
+					{
+						Name:        "test-hook-2",
+						PlaceHookOn: "ignite chain",
+					},
+				},
+			},
+			shouldError: true,
+		},
+		{
+			name: "ok: should prepend root command",
+			pluginInterface: pluginInterface{
+				commands: []plugin.Command{
+					{
+						Use: "foo",
+					},
+				},
+				hooks: []plugin.Hook{
+					{
+						Name:        "test-hook-3",
+						PlaceHookOn: "scaffold chain",
+					},
+				},
+			},
+			shouldError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			require := require.New(t)
+			assert := assert.New(t)
+			p := &plugin.Plugin{
+				Plugin:    chainconfig.Plugin{Path: "foo"},
+				Interface: tt.pluginInterface,
+			}
+			rootCmd := buildRootCmd()
+
+			linkPluginHooks(rootCmd, p)
+			if tt.shouldError {
+				require.Error(p.Error)
+			} else {
+				require.NoError(p.Error)
+			}
+
+			for _, hook := range p.Interface.Hooks() {
+				areHooksDefined := checkCmd(rootCmd, hook.PlaceHookOn)
+				if tt.shouldError {
+					assert.False(areHooksDefined, "hooks should have pre and post runs undefined")
+				} else {
+					assert.True(areHooksDefined, "hooks should have pre and post runs defined")
+				}
+			}
+		})
+	}
```

Comment:
this test fails on my machine.

Reply:
Can you send me the failure case and stack? It works on my machine (M1 MacBook Pro) but that's not very helpful.